gaphor: 3.0.0 → 3.1.0; overhaul derivation builder

normalcea opened the pull request:
Paging issue #291281 and PR #378026 as well as PR #387263, this updates gaphor to the latest release and moves it into `pkgs/by-name`

Notable changes include replacing the Pypi source with the GitHub one instead for additional files as well as overhauling much of the builder code and meta attrs.

This PR is still not finished as gaphor does not integrate with gsettings at all and complains at startup about missing gsettings files. It's just nearly at the finish line.

Section in question, I'm pretty sure this might also need a guard for cross-comp as well, I mostly took this from https://github.com/flathub/org.gaphor.Gaphor flatpak builder.
```bash
   $out/bin/gaphor install-schemas \
                    --schemas-dir $out/share/glib-2.0/schemas \
                    --glib-compile-schemas ${glib.dev}/bin/glib-compile-schemas
```

I also would like to open an issue upstream about using meson as their build system and I'll find the time to do so. Otherwise, huge `postInstall` block. 

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)***
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Commit:
gaphor: move to pkgs/by-name

Commit:
gaphor: furnish meta attributes; add normalcea as maintainer

Commit:
gaphor: 3.0.0 → 3.1.0

normalcea commented:
force pushed to visually flatten `postInstall` phase.

normalcea commented on pkgs/by-name/ga/gaphor/package.nix:
```diff
@@ -0,0 +1,113 @@
+
+    install -Dm644 data/org.gaphor.Gaphor.xml -t $out/share/mime/packages/
+    install -Dm644 data/logos/gaphor-24x24.png -t $out/share/icons/hicolor/24x24/apps/
+    install -Dm644 data/logos/gaphor-48x48.png -t $out/share/icons/hicolor/48x48/apps/
```
Fixed to match what Flathub uses[1].

- [1] https://github.com/flathub/org.gaphor.Gaphor/blob/master/org.gaphor.Gaphor.yaml

normalcea commented on pkgs/by-name/ga/gaphor/package.nix:
```diff
@@ -0,0 +1,113 @@
+
+    install -Dm644 data/org.gaphor.Gaphor.desktop -t $out/share/applications/
+    substituteInPlace $out/share/applications/org.gaphor.Gaphor.desktop \
+                      --replace-fail "Exec=gaphor" "Exec=$out/bin/gaphor"
```
I'm going to shelve that PR in my notes now. 

And speaking of PRs, I want to raise the issue of using the meson build-system since that's been the de-facto norm for Adwaita applications.

Aleksanaa commented on pkgs/by-name/ga/gaphor/package.nix:
```diff
@@ -0,0 +1,113 @@
+    install -Dm644 data/logos/org.gaphor.Gaphor-symbolic.svg -t  \
+                   $out/share/icons/hicolor/symbolic/apps/
+
+    $out/bin/gaphor install-schemas \
+                    --schemas-dir $out/share/glib-2.0/schemas \
+                    --glib-compile-schemas ${glib.dev}/bin/glib-compile-schemas
```
This breaks cross compilation, but I guess we cannot avoid it.

And even if it is installed, it still complaints about not finding schema:

```
=> ./results/gaphor-x86_64-linux/bin/gaphor 
gaphor.settings WARNING Settings schema not found and settings won’t be saved. Run `gaphor install-schemas`.
```

Aleksanaa replied:
https://github.com/gaphor/gaphor/blob/94ff98d5fe95b6f3b69e9ae03d5ec944041d74ca/gaphor/ui/installschemas/__init__.py#L18
https://github.com/gaphor/gaphor/blob/94ff98d5fe95b6f3b69e9ae03d5ec944041d74ca/gaphor/ui/installschemas/__init__.py#L41

It looks simple, so we would better install these ourselves.

normalcea commented:
- Rebased on current master
- Replaced `gaphor install-schemas` with manual installation:

```bash
install -Dm644 gaphor/ui/installschemas/org.gaphor.Gaphor.gschema.xml -t $out/share/glib-2.0/schemas/
glib-compile-schemas $out/share/glib-2.0/schemas/
```
- Changed styling of the install commands to be flat and rearranged order implicitly. 

I don't know if the lines should be split or not since it is very wide, so guidance on that would be applicable.

Aleksanaa commented:
Still

```
gaphor.settings WARNING Settings schema not found and settings won’t be saved. Run `gaphor install-schemas`.
```

https://github.com/gaphor/gaphor/blob/94ff98d5fe95b6f3b69e9ae03d5ec944041d74ca/gaphor/settings.py#L61

normalcea commented:
As well, changing the theme of the application doesn't work, I'll test the gaphor flatpak and see if how reproducible this schema issue  there in the meantime. 

They should really be using meson to standardize this :disappointed:

normalcea commented:
Schemas also don't function for me even if they're installed locally in `~/.local/share/glib-2.0/schemas/`

normalcea commented:
I'm still stumped by this schema issue, since there are only like 3-4 schemas, would it be worthwhile to merge this PR which fixes building and deal with schemas and build-system upstream?
